core/server/api/configuration.js

```diff
@@ -54,9 +61,29 @@ configuration = {
      */
     read: function read(options) {
         options = options || {};
+        var ops = {};
 
         if (!options.key) {
-            return Promise.resolve({configuration: [getBaseConfig()]});
+            ops.ghostAdmin = models.Client.findOne({slug: 'ghost-admin'});
+
+            if (config.get('auth:type') === 'ghost') {
+                ops.ghostAuth = models.Client.findOne({slug: 'ghost-auth'});
+            }
+
+            return Promise.props(ops)
+                .then(function (result) {
+                    var configuration = getBaseConfig();
+
+                    configuration.clientId = {value: result.ghostAdmin.get('slug'), type: 'string'};
+                    configuration.clientSecret = {value: result.ghostAdmin.get('secret'), type: 'string'};
+
+                    if (result.ghostAuth) {
+                        configuration.ghostAuthId = {value: result.ghostAuth.get('uuid'), type: 'string'};
```

Comment:
I don't think this one should be added to the public endpoint - the auth ID should only be known by the server as it's used by the oauth server to identify that it's the ghost server talking to it unless I'm misunderstanding what `result.ghostAuth.get('uuid')` resolves to